### PR TITLE
feat(ai): implement async sdk requests in toolkit

### PR DESCRIPTION
**Pull request by mfadler**

### CODE REVIEW

File: unique_toolkit/unique_toolkit/_common/_common_service.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import logging
+from typing import Optional
+
+from unique_toolkit.chat.state import ChatState
+
+
+class CommonService:
```
Comment (sheiksadique) [inline unique_toolkit/unique_toolkit/_common/_common_service.py:7]: Perhaps call this a `BaseService` ?

File: unique_toolkit/pyproject.toml [hunk 2]
```diff
@@ -19,9 +19,10 @@ pyhumps = "^3.8.0"
 unique-sdk = "^0.9.2"
 numpy = "^1.26.4"
 python-dotenv = "^1.0.1"
-requests = "^2.32.3"
 regex = "^2024.5.15"
 tiktoken = "^0.7.0"
+requests = "^2.32.3"
+httpx = "^0.27.0"
```
Comment (sheiksadique) [inline unique_toolkit/pyproject.toml:25]: If httpx is part of the requirements, the other implementations will never really be used right? Is there a way to explicitly select which backend to use?

File: unique_toolkit/unique_toolkit/content/service.py [hunk 3]
```diff
@@ -41,32 +49,50 @@ def search_content_chunks(
...
+            searches = unique_sdk.Search.create(
+                user_id=self.state.user_id,
+                company_id=self.state.company_id,
+                chatId=self.state.chat_id,
+                searchString=search_string,
+                searchType=search_type.name,
+                scopeIds=scope_ids,
+                limit=limit,
+                reranker=reranker_config,  # type: ignore
+                language=search_language,  # type: ignore
```
Comment (sheiksadique) [inline unique_toolkit/unique_toolkit/content/service.py:73]: You should now be able to remove the type ignore statements with unique_sdk >=0.9.3.

File: unique_toolkit/unique_toolkit/language_model/service.py [hunk 4]
```diff
@@ -44,22 +51,32 @@ def complete(
...
+                company_id=self.state.company_id,
+                # TODO change or extend types in unique_sdk
+                model=model_name.name,  # type: ignore
+                messages=cast(
+                    list[unique_sdk.Integrated.ChatCompletionRequestMessage],
+                    messages,
+                ),
+                timeout=timeout,
+                temperature=temperature,
+                options=options,  # type: ignore
```
Comment (sheiksadique) [inline unique_toolkit/unique_toolkit/language_model/service.py:67]: There is no explicit mention of `options` in the `unique_sdk`. @unique-fabian are you aware of what this does and if it is needed?

File: unique_toolkit/unique_toolkit/language_model/service.py [hunk 5]
```diff
@@ -106,20 +107,19 @@ def _trigger_complete(
                 temperature=temperature,
                 options=options,  # type: ignore
```
Comment (sheiksadique) [inline unique_toolkit/unique_toolkit/language_model/service.py:108]: same as above. Is options actually consumed by the API?

File: unique_toolkit/pyproject.toml (hunk 2 above)
Comment (mfadler) [reply unique_toolkit/pyproject.toml]: I removed the dependencies to keep it consistent with the usage of the sdk. But there is the possibility to explicit define a client by setting `unique_sdk.default_http_client` but u cannot change the version.

Maybe we add a `unique_sdk.default_async_http_client` to keep sync and async separate.

File: unique_toolkit/unique_toolkit/language_model/service.py (hunk 4 above)
Comment (mfadler) [reply unique_toolkit/unique_toolkit/language_model/service.py:67]: I discussed this with Fabian and also wondered that the parameter is not present in the sdk but somehow it works. ^^

File: unique_toolkit/unique_toolkit/language_model/service.py (hunk 5 above)
Comment (mfadler) [reply unique_toolkit/unique_toolkit/language_model/service.py:108]: see answer above

File: unique_toolkit/unique_toolkit/content/service.py [hunk 6]
```diff
@@ -75,46 +101,30 @@ def async_search_content_chunks(
...
                 searchString=search_string,
                 searchType=search_type.name,
                 scopeIds=scope_ids,
                 limit=limit,
-                reranker=reranker_config.model_dump() if reranker_config else None,
-                language=search_language,
-                chatOnly=self.state.chat_only,
+                # TODO fix types in SDK
+                reranker=reranker_config.model_dump() if reranker_config else None,  # type: ignore
+                language=search_language,  # type: ignore
```
[the remark this answers is not in the document]
Comment (mfadler) [reply unique_toolkit/unique_toolkit/content/service.py:126]: bumped up sdk version and removed comments

File: unique_toolkit/unique_toolkit/content/service.py [hunk 7]
```diff
@@ -18,21 +18,29 @@
...
     def search_content_chunks(
         self,
         search_string: str,
         search_type: ContentSearchType,
         limit: int,
-        reranker_config: Optional[RerankerConfig] = None,
         search_language: str = DEFAULT_SEARCH_LANGUAGE,
-        scope_ids: Optional[list[str]] = None,
+        reranker_config: Optional[RerankerConfig] = None,
+        scope_ids: Optional[list[str]] = [],
```
Comment (sheiksadique) [inline unique_toolkit/unique_toolkit/content/service.py:42]: Default value here should be `None`. 
It is also not a good idea to assign a `[]` and `{}` as default in methods. (https://www.pullrequest.com/blog/python-pitfalls-the-perils-of-using-lists-and-dicts-as-default-arguments/)

File: unique_toolkit/unique_toolkit/content/service.py [hunk 8]
```diff
@@ -41,32 +49,50 @@ def search_content_chunks(
...
+    async def search_content_chunks_async(
         self,
         search_string: str,
         search_type: ContentSearchType,
         limit: int,
-        reranker_config: Optional[RerankerConfig] = None,
         search_language: str = DEFAULT_SEARCH_LANGUAGE,
-        scope_ids: Optional[list[str]] = None,
+        reranker_config: Optional[RerankerConfig] = None,
+        scope_ids: Optional[list[str]] = [],
```
Comment (sheiksadique) [inline unique_toolkit/unique_toolkit/content/service.py:94]: Same as above. Please change to `None`.

File: unique_toolkit/unique_toolkit/language_model/service.py (hunk 4 above)
Comment (sheiksadique) [reply unique_toolkit/unique_toolkit/language_model/service.py:67]: K thankyou. Let's keep this in mind and add it to the `sdk` explicitly.

**Review by sheiksadique — APPROVED** [review]
lgtm

**Pull request merged**